custom_components/tuya_local/devices/agl_ultramagic_lock.yaml

```diff
@@ -1,99 +1,56 @@
-name: AGL Ultra Magic lock
+name: AGL Ultra Magic Lock
```

Comment:
Move the branding into the name under products, and make this something generic.  Maybe "Gate opener" would be more suitable, since it doesn't contain a lock entity any more.